### PR TITLE
Cancel previous running workflows when a new commit is made

### PR DESCRIPTION
**Pull request by ceyhun**
## Description

Added a new GitHub action workflow named `Cancel` using [styfle/cancel-workflow-action](https://github.com/styfle/cancel-workflow-action) to cancel all running workflows on a branch when a new commit is made.

## How has this been tested?

See the commits `Test 1` and `Test 2` cancelling previous runs. Force-pushes also work.

### CODE REVIEW

**Comment by gziolo** [comment]
I know little about GitHub actions. I can approve just to validate it in action. @youknowriad what do you think?

File: .github/workflows/cancel.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: Cancel
+on: [push]
```
Comment (youknowriad) [inline .github/workflows/cancel.yml:2]: push only runs if you have the rights on the repository, not if you make a PR from a fork. How can we ensure to run this on fork PRs updates too?

**Comment by youknowriad** [comment]
Github should probably do this by default but this is cool in the meantime.

File: .github/workflows/cancel.yml (hunk 1 above)
Comment (ceyhun) [reply .github/workflows/cancel.yml]: I changed it to trigger only on `pull_request` events, which should run on forked repos as well: https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-events-for-forked-repositories

Most of the current workflows are already triggered on `pull_request` and some also on `push` to `master`. This way it won't cancel those triggered via `push` to `master` and maybe that's better.

**Review by youknowriad — APPROVED** [review]
Let's try it.

**Comment by youknowriad** [comment]
This job is cool but it triggers too many email notifications for me so i end up missing the good notifications. Any way we can disable notifications just for this job?

**Comment by ceyhun** [comment]
What kind of notifications? Do you mean ones with titles starting with `[WordPress/gutenberg] Run cancelled...` for the cancelled jobs?

**Comment by youknowriad** [comment]
yes, these notifications. I can filter in my own client but I'm pretty sure it's an annoyance for most folks. 
i guess we might not be able to do anything about it because it's triggered by the test jobs themselves.

**Comment by ceyhun** [comment]
> i guess we might not be able to do anything about it because it's triggered by the test jobs themselves.

Yeah I think that's the case, maybe we can send feedback about it using this form: https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions

**Pull request merged**